src/main/java/bio/terra/cli/service/SamService.java

```diff
@@ -66,7 +66,10 @@ public static SamService fromContext() {
 
   /** Factory method for class that talks to SAM. Pulls the current server from the context. */
   public static SamService forUser(User user) {
-    return new SamService(user.getUserAccessToken(), Context.getServer());
+    Server server = Context.getServer();
+    AccessToken token =
```

Comment:
This is an odd access pattern, since you're asking one object which method to call on another object. It's almost like the server knows too much.